### PR TITLE
tiup: remove a false download link for tiup offline package

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Remove the link of [Download TiDB | PingCAP](https://en.pingcap.com/download/) in `production-offline-deployment-using-tiup.md` because the [Download TiDB | PingCAP](https://en.pingcap.com/download/) page is designed not to provide the download link for TiUP offline package.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by King-Dylan — APPROVED** [review]
lgtm

File: production-offline-deployment-using-tiup.md [hunk 1]
```diff
@@ -14,13 +14,7 @@ This document describes how to deploy a TiDB cluster offline using TiUP.
...
-### Option 1: Download the official TiUP offline component package
-
-Download the offline mirror package of the TiDB server (including the TiUP offline component package) from the [Download TiDB](https://pingcap.com/download/) page.
-
-### Option 2: Manually pack an offline component package using `tiup mirror clone`
-
-The steps are below.
+To prepare the TiUP offline component package, manually pack an offline component package using `tiup mirror clone`.
 
 - Install the TiUP package manager online.
```
Comment (lilin90) [inline production-offline-deployment-using-tiup.md:19]: It seems that this section describes the ordered steps.
```suggestion
1. Install the TiUP package manager online.
```

File: production-offline-deployment-using-tiup.md [hunk 2]
```diff
@@ -48,7 +42,7 @@ The steps are below.
         which tiup
         ```
 
-- Pull the mirror using TiUP
+2. Pull the mirror using TiUP
```
Comment (TomShawn) [inline production-offline-deployment-using-tiup.md:45]:
```suggestion
2. Pull the mirror using TiUP.
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Pull request merged**